### DISCUSSION TITLE
AutoProtoSchemaBuilderAnnotationProcessor: Unable to get public no-arg constructor

### DISCUSSION THREAD

**Discussion by nikolas-charalambidis** [infinispan/infinispan#12953] [Q&A] [answered]
I am compiling a project, crashing on the following error:
```[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.0:compile (default-compile) on project thub-demo: Compilation failure
[ERROR] Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: org.infinispan.protostream.annotations.impl.processor.AutoProtoSchemaBuilderAnnotationProcessor Unable to get public no-arg constructor
```

- Spring Boot 3.3.4
- Infinispan 15.0.8.Final
   ```
   <dependency>
            <groupId>org.infinispan</groupId>
            <artifactId>infinispan-spring-boot3-starter-embedded</artifactId>
            <scope>runtime</scope>
    </dependency>
    ```

Does anyone have a clue what can cause it?

**Comment by tristantarrant** [comment]
It's odd because `AutoProtoSchemaBuilderAnnotationProcessor` doesn't have an explicitly declared constructor, therefore it will have a default no-arg one. What JDK does this happen with ?

**Reply by nikolas-charalambidis** [reply]
Exactly, the compiler creates a public no-args constructor. I have no clue where to start digging and what dependencies might collide (though I use Spring dependencies BOM so I have no conflicts).

I use Java 17:
```
17.0.1 (x86_64) "Oracle Corporation" - "OpenJDK 17.0.1" /Library/Java/JavaVirtualMachines/jdk-17.0.1.jdk/Contents/Home
```

**Reply by tristantarrant** [reply]
I'll try and replicate that with that exact version of the JDK. 
However, you should really upgrade your JDK to a newer one like 17.0.11.

**Reply by tristantarrant** [reply, marked as answer]
Another problem is with `<scope>runtime</scope>`.
According to https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope

> **runtime**
> This scope indicates that the dependency is not required for compilation, but is for execution. Maven includes a dependency with this scope in the runtime and test classpaths, but not the compile classpath.

**Reply by nikolas-charalambidis** [reply]
@tristantarrant Thanks for the investigation. There are two ways to overcome the issue:

1. As you said, include the dependency in the default `compile` scope rather than `runtime`. 
    ```xml
    <dependency>
         <groupId>org.infinispan</groupId>
         <artifactId>infinispan-spring-boot3-starter-embedded</artifactId>
         <scope>runtime</scope>
     </dependency>
    ```

2. Disable annotation processing in the Maven Compiler plugin:
    ```xml
    <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
         <version>3.12.0</version>
         <configuration>
            <proc>none</proc>
         </configuration>
     </plugin>
    ```

I guess the first one is rather what I want.
